<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Publicly available datasets | The Bacterial Genomics Book</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.44 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Publicly available datasets | The Bacterial Genomics Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Publicly available datasets | The Bacterial Genomics Book" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Amber Baton" />


<meta name="date" content="2025-09-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="running-commands-on-many-samples.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Learning the basics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-command-line"><i class="fa fa-check"></i><b>1.1</b> The Command Line</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r"><i class="fa fa-check"></i><b>1.2</b> R</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#python"><i class="fa fa-check"></i><b>1.3</b> Python</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#interactive-tools"><i class="fa fa-check"></i><b>1.4</b> Interactive tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html"><i class="fa fa-check"></i><b>2</b> Publicly available datasets</a>
<ul>
<li class="chapter" data-level="2.1" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html#downloading-fastq-files-from-a-published-study"><i class="fa fa-check"></i><b>2.1</b> Downloading fastq files from a published study</a></li>
<li class="chapter" data-level="2.2" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html#downloading-fastq-files-from-a-systematic-search-or-multi-study-collection"><i class="fa fa-check"></i><b>2.2</b> Downloading fastq files from a systematic search or multi-study collection</a></li>
<li class="chapter" data-level="2.3" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html#downloading-assemblies-from-online-databases"><i class="fa fa-check"></i><b>2.3</b> Downloading assemblies from online databases</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html"><i class="fa fa-check"></i><b>3</b> Running commands on many samples</a>
<ul>
<li class="chapter" data-level="3.1" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html#loops"><i class="fa fa-check"></i><b>3.1</b> For loops</a></li>
<li class="chapter" data-level="3.2" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html#while-loops"><i class="fa fa-check"></i><b>3.2</b> While loops</a></li>
<li class="chapter" data-level="3.3" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html#write-a-script"><i class="fa fa-check"></i><b>3.3</b> Write a script</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html"><i class="fa fa-check"></i><b>4</b> Fastq quality control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html#fastqc-and-multiqc"><i class="fa fa-check"></i><b>4.1</b> FastQC and MultiQC</a></li>
<li class="chapter" data-level="4.2" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html#trimming-reads"><i class="fa fa-check"></i><b>4.2</b> Trimming reads</a></li>
<li class="chapter" data-level="4.3" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html#contamination"><i class="fa fa-check"></i><b>4.3</b> Contamination</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genome-assembly.html"><a href="genome-assembly.html"><i class="fa fa-check"></i><b>5</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-assembly.html"><a href="genome-assembly.html#spades"><i class="fa fa-check"></i><b>5.1</b> Short-read</a></li>
<li class="chapter" data-level="5.2" data-path="genome-assembly.html"><a href="genome-assembly.html#hybrid-and-long-read"><i class="fa fa-check"></i><b>5.2</b> Hybrid and long-read</a></li>
<li class="chapter" data-level="5.3" data-path="genome-assembly.html"><a href="genome-assembly.html#quality-control"><i class="fa fa-check"></i><b>5.3</b> Quality control</a></li>
<li class="chapter" data-level="5.4" data-path="genome-assembly.html"><a href="genome-assembly.html#prokka"><i class="fa fa-check"></i><b>5.4</b> Annotation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="alignments.html"><a href="alignments.html"><i class="fa fa-check"></i><b>6</b> Alignments</a>
<ul>
<li class="chapter" data-level="6.1" data-path="alignments.html"><a href="alignments.html#refbased"><i class="fa fa-check"></i><b>6.1</b> Reference-based alignments</a></li>
<li class="chapter" data-level="6.2" data-path="alignments.html"><a href="alignments.html#coverage"><i class="fa fa-check"></i><b>6.2</b> Checking coverage</a></li>
<li class="chapter" data-level="6.3" data-path="alignments.html"><a href="alignments.html#sub-setting-an-alignment"><i class="fa fa-check"></i><b>6.3</b> Sub-setting an alignment</a></li>
<li class="chapter" data-level="6.4" data-path="alignments.html"><a href="alignments.html#detecting-recombination"><i class="fa fa-check"></i><b>6.4</b> Detecting recombination</a></li>
<li class="chapter" data-level="6.5" data-path="alignments.html"><a href="alignments.html#masking-regions"><i class="fa fa-check"></i><b>6.5</b> Masking regions</a></li>
<li class="chapter" data-level="6.6" data-path="alignments.html"><a href="alignments.html#coregene"><i class="fa fa-check"></i><b>6.6</b> Core gene alignments</a></li>
<li class="chapter" data-level="6.7" data-path="alignments.html"><a href="alignments.html#snp-alignment"><i class="fa fa-check"></i><b>6.7</b> SNP alignment</a></li>
<li class="chapter" data-level="6.8" data-path="alignments.html"><a href="alignments.html#vcf-conversion"><i class="fa fa-check"></i><b>6.8</b> VCF conversion</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="building-trees.html"><a href="building-trees.html"><i class="fa fa-check"></i><b>7</b> Building trees</a>
<ul>
<li class="chapter" data-level="7.1" data-path="building-trees.html"><a href="building-trees.html#fasttree"><i class="fa fa-check"></i><b>7.1</b> FastTree</a></li>
<li class="chapter" data-level="7.2" data-path="building-trees.html"><a href="building-trees.html#iq-tree"><i class="fa fa-check"></i><b>7.2</b> IQ-TREE</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="rooting-trees.html"><a href="rooting-trees.html"><i class="fa fa-check"></i><b>8</b> Rooting trees</a>
<ul>
<li class="chapter" data-level="8.1" data-path="rooting-trees.html"><a href="rooting-trees.html#outgroup"><i class="fa fa-check"></i><b>8.1</b> Outgroup</a></li>
<li class="chapter" data-level="8.2" data-path="rooting-trees.html"><a href="rooting-trees.html#midpoint"><i class="fa fa-check"></i><b>8.2</b> Midpoint</a></li>
<li class="chapter" data-level="8.3" data-path="rooting-trees.html"><a href="rooting-trees.html#molecular-clock"><i class="fa fa-check"></i><b>8.3</b> Molecular clock</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="rooting-trees.html"><a href="rooting-trees.html#bactdating"><i class="fa fa-check"></i><b>8.3.1</b> BactDating</a></li>
<li class="chapter" data-level="8.3.2" data-path="rooting-trees.html"><a href="rooting-trees.html#treetime"><i class="fa fa-check"></i><b>8.3.2</b> TreeTime</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bacterial Genomics Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="publicly-available-datasets" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Publicly available datasets<a href="publicly-available-datasets.html#publicly-available-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="downloading-fastq-files-from-a-published-study" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Downloading fastq files from a published study<a href="publicly-available-datasets.html#downloading-fastq-files-from-a-published-study" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For sequencing papers, often a Study Accession (e.g. PRJEB8764) will be provided in the methods, and sample or run accessions (e.g. ERR1880809) in the supplementary documents. One of the easiest ways of downloading fastq files is to visit the <a href="https://www.ebi.ac.uk/ena/browser/home">ENA website</a>, type in <a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB8764">an accession</a>, and click “download”. If you click “download all”, you will be given a script you can run in the command line to download the files at scale.</p>
</div>
<div id="downloading-fastq-files-from-a-systematic-search-or-multi-study-collection" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Downloading fastq files from a systematic search or multi-study collection<a href="publicly-available-datasets.html#downloading-fastq-files-from-a-systematic-search-or-multi-study-collection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Go on to <a href="https://www.ncbi.nlm.nih.gov/sra">NCBI SRA</a> and type in your search. For example, if I wanted to find Shigella (taxonomic ID 624) DNA whole-genome sequences from India, I might type in:</p>
<p><strong>txid624[Organism] AND “WGS”[Strategy] AND “biomol dna”[Properties] AND “India”[Country]</strong></p>
<p>You can then click “Send results to Run selector”, which will give you a neatly formatted table with a list of the run accessions and metadata. Once you have a list of run accessions, you can download the fastq files using <a href="https://hpc.nih.gov/apps/sratoolkit.html">SRA toolkit</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="publicly-available-datasets.html#cb1-1" tabindex="-1"></a><span class="co">#Downloads runs in SRA format</span></span>
<span id="cb1-2"><a href="publicly-available-datasets.html#cb1-2" tabindex="-1"></a><span class="ex">prefetch</span> <span class="at">--option-file</span> SraAccList.txt</span>
<span id="cb1-3"><a href="publicly-available-datasets.html#cb1-3" tabindex="-1"></a><span class="co">#Loop through the downloaded files to convert to fastq</span></span>
<span id="cb1-4"><a href="publicly-available-datasets.html#cb1-4" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="pp">*</span>RR<span class="pp">*</span></span>
<span id="cb1-5"><a href="publicly-available-datasets.html#cb1-5" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-6"><a href="publicly-available-datasets.html#cb1-6" tabindex="-1"></a><span class="ex">fasterq-dump</span> <span class="at">--split-files</span> <span class="va">$i</span></span>
<span id="cb1-7"><a href="publicly-available-datasets.html#cb1-7" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-8"><a href="publicly-available-datasets.html#cb1-8" tabindex="-1"></a><span class="co">#Compress to get fastq.gz files</span></span>
<span id="cb1-9"><a href="publicly-available-datasets.html#cb1-9" tabindex="-1"></a><span class="fu">gzip</span> <span class="pp">*</span>fastq</span></code></pre></div>
<p>Alternatively, the following R function, which uses the readR and httr packages, can be used to retrieve the url of a fastq.gz file when given a run accession.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="publicly-available-datasets.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(httr)</span>
<span id="cb2-2"><a href="publicly-available-datasets.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb2-3"><a href="publicly-available-datasets.html#cb2-3" tabindex="-1"></a>get_fastq_urls <span class="ot">&lt;-</span> <span class="cf">function</span>(accession) {</span>
<span id="cb2-4"><a href="publicly-available-datasets.html#cb2-4" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">GET</span>(<span class="st">&quot;https://www.ebi.ac.uk/ena/portal/api/filereport&quot;</span>,</span>
<span id="cb2-5"><a href="publicly-available-datasets.html#cb2-5" tabindex="-1"></a>             <span class="at">query =</span> <span class="fu">list</span>(<span class="at">accession =</span> accession, <span class="at">result =</span> <span class="st">&quot;read_run&quot;</span>,</span>
<span id="cb2-6"><a href="publicly-available-datasets.html#cb2-6" tabindex="-1"></a>                          <span class="at">fields =</span> <span class="st">&quot;fastq_ftp&quot;</span>, <span class="at">format =</span> <span class="st">&quot;tsv&quot;</span>))</span>
<span id="cb2-7"><a href="publicly-available-datasets.html#cb2-7" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">content</span>(res, <span class="st">&quot;text&quot;</span>, <span class="at">encoding =</span> <span class="st">&quot;UTF-8&quot;</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-8"><a href="publicly-available-datasets.html#cb2-8" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(df) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">||</span> <span class="sc">!</span><span class="st">&quot;fastq_ftp&quot;</span> <span class="sc">%in%</span> <span class="fu">names</span>(df) <span class="sc">||</span> <span class="fu">is.na</span>(df<span class="sc">$</span>fastq_ftp) <span class="sc">||</span> df<span class="sc">$</span>fastq_ftp <span class="sc">==</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb2-9"><a href="publicly-available-datasets.html#cb2-9" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">warning</span>(<span class="fu">sprintf</span>(<span class="st">&quot;No FASTQ found for %s&quot;</span>, accession)))</span>
<span id="cb2-10"><a href="publicly-available-datasets.html#cb2-10" tabindex="-1"></a>  <span class="fu">paste0</span>(<span class="st">&quot;https://&quot;</span>, <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">as.character</span>(df<span class="sc">$</span>fastq_ftp), <span class="st">&quot;;&quot;</span>)))}</span>
<span id="cb2-11"><a href="publicly-available-datasets.html#cb2-11" tabindex="-1"></a></span>
<span id="cb2-12"><a href="publicly-available-datasets.html#cb2-12" tabindex="-1"></a><span class="fu">get_fastq_urls</span>(<span class="st">&quot;ERR1880809&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR188/009/ERR1880809/ERR1880809_1.fastq.gz&quot;
## [2] &quot;https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR188/009/ERR1880809/ERR1880809_2.fastq.gz&quot;</code></pre>
<p>You can then download them using the wget function in the command line.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="publicly-available-datasets.html#cb4-1" tabindex="-1"></a><span class="fu">wget</span> https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR188/009/ERR1880809/ERR1880809_1.fastq.gz</span>
<span id="cb4-2"><a href="publicly-available-datasets.html#cb4-2" tabindex="-1"></a><span class="fu">wget</span> https://ftp.sra.ebi.ac.uk/vol1/fastq/ERR188/009/ERR1880809/ERR1880809_2.fastq.gz</span></code></pre></div>
</div>
<div id="downloading-assemblies-from-online-databases" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Downloading assemblies from online databases<a href="publicly-available-datasets.html#downloading-assemblies-from-online-databases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="%22https://enterobase.warwick.ac.uk/%22">Enterobase</a> and <a href="%22https://pathogen.watch/%22">Pathogenwatch</a> are both good resources for downloading ready-made assemblies for certain pathogens, particularly enteric pathogens.</p>
<p>For a more comprehensive search, you could visit <a href="https://www.ncbi.nlm.nih.gov/datasets/genome/">NCBI Genome</a>. Currently, if I type in “Vibrio cholerae”, NCBI genome will return 14,025 genomes. If I click <em>filter</em> I can then filter by assembly level, for example “chromosome” (516 genomes) or “complete” (474 genomes). These genomes can be downloaded using <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/how-tos/genomes/download-genome/">ncbi-datasets-cli</a>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="publicly-available-datasets.html#cb5-1" tabindex="-1"></a><span class="co">#Download all complete Vibrio cholerae assemblies</span></span>
<span id="cb5-2"><a href="publicly-available-datasets.html#cb5-2" tabindex="-1"></a><span class="ex">datasets</span> download genome taxon 666 <span class="at">--assembly-level</span> complete <span class="at">--filename</span> Cholera.zip <span class="at">--include</span> genome</span>
<span id="cb5-3"><a href="publicly-available-datasets.html#cb5-3" tabindex="-1"></a><span class="co">#Download a particular assembly for which you have an accession</span></span>
<span id="cb5-4"><a href="publicly-available-datasets.html#cb5-4" tabindex="-1"></a><span class="ex">datasets</span> download genome accession GCA_003574155.1 <span class="at">--filename</span> V060002.zip <span class="at">--include</span> genome</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="running-commands-on-many-samples.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-intro.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
