<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Genome assembly | The Bacterial Genomics Book</title>
  <meta name="description" content="This book outlines basic analyses for bacterial genomics." />
  <meta name="generator" content="bookdown 0.44 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Genome assembly | The Bacterial Genomics Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book outlines basic analyses for bacterial genomics." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Genome assembly | The Bacterial Genomics Book" />
  
  <meta name="twitter:description" content="This book outlines basic analyses for bacterial genomics." />
  

<meta name="author" content="Amber Barton" />


<meta name="date" content="2025-10-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="fastq-quality-control.html"/>
<link rel="next" href="alignments.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The Bacterial Genomics Book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Learning the basics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-command-line"><i class="fa fa-check"></i><b>1.1</b> The Command Line</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r"><i class="fa fa-check"></i><b>1.2</b> R</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#python"><i class="fa fa-check"></i><b>1.3</b> Python</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#interactive-tools"><i class="fa fa-check"></i><b>1.4</b> Interactive tools</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html"><i class="fa fa-check"></i><b>2</b> Publicly available datasets</a>
<ul>
<li class="chapter" data-level="2.1" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html#downloading-fastq-files-from-a-published-study"><i class="fa fa-check"></i><b>2.1</b> Downloading fastq files from a published study</a></li>
<li class="chapter" data-level="2.2" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html#downloading-fastq-files-from-a-systematic-search-or-multi-study-collection"><i class="fa fa-check"></i><b>2.2</b> Downloading fastq files from a systematic search or multi-study collection</a></li>
<li class="chapter" data-level="2.3" data-path="publicly-available-datasets.html"><a href="publicly-available-datasets.html#downloading-assemblies-from-online-databases"><i class="fa fa-check"></i><b>2.3</b> Downloading assemblies from online databases</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html"><i class="fa fa-check"></i><b>3</b> Running commands on many samples</a>
<ul>
<li class="chapter" data-level="3.1" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html#loops"><i class="fa fa-check"></i><b>3.1</b> For loops</a></li>
<li class="chapter" data-level="3.2" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html#while-loops"><i class="fa fa-check"></i><b>3.2</b> While loops</a></li>
<li class="chapter" data-level="3.3" data-path="running-commands-on-many-samples.html"><a href="running-commands-on-many-samples.html#write-a-script"><i class="fa fa-check"></i><b>3.3</b> Write a script</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html"><i class="fa fa-check"></i><b>4</b> Fastq quality control</a>
<ul>
<li class="chapter" data-level="4.1" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html#fastqc-and-multiqc"><i class="fa fa-check"></i><b>4.1</b> FastQC and MultiQC</a></li>
<li class="chapter" data-level="4.2" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html#trimming-reads"><i class="fa fa-check"></i><b>4.2</b> Trimming reads</a></li>
<li class="chapter" data-level="4.3" data-path="fastq-quality-control.html"><a href="fastq-quality-control.html#contamination"><i class="fa fa-check"></i><b>4.3</b> Contamination</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="genome-assembly.html"><a href="genome-assembly.html"><i class="fa fa-check"></i><b>5</b> Genome assembly</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-assembly.html"><a href="genome-assembly.html#spades"><i class="fa fa-check"></i><b>5.1</b> Short-read</a></li>
<li class="chapter" data-level="5.2" data-path="genome-assembly.html"><a href="genome-assembly.html#hybrid-and-long-read"><i class="fa fa-check"></i><b>5.2</b> Hybrid and long-read</a></li>
<li class="chapter" data-level="5.3" data-path="genome-assembly.html"><a href="genome-assembly.html#quality-control"><i class="fa fa-check"></i><b>5.3</b> Quality control</a></li>
<li class="chapter" data-level="5.4" data-path="genome-assembly.html"><a href="genome-assembly.html#prokka"><i class="fa fa-check"></i><b>5.4</b> Annotation</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="alignments.html"><a href="alignments.html"><i class="fa fa-check"></i><b>6</b> Alignments</a>
<ul>
<li class="chapter" data-level="6.1" data-path="alignments.html"><a href="alignments.html#refbased"><i class="fa fa-check"></i><b>6.1</b> Reference-based alignments</a></li>
<li class="chapter" data-level="6.2" data-path="alignments.html"><a href="alignments.html#coverage"><i class="fa fa-check"></i><b>6.2</b> Checking coverage</a></li>
<li class="chapter" data-level="6.3" data-path="alignments.html"><a href="alignments.html#sub-setting-an-alignment"><i class="fa fa-check"></i><b>6.3</b> Sub-setting an alignment</a></li>
<li class="chapter" data-level="6.4" data-path="alignments.html"><a href="alignments.html#detecting-recombination"><i class="fa fa-check"></i><b>6.4</b> Detecting recombination</a></li>
<li class="chapter" data-level="6.5" data-path="alignments.html"><a href="alignments.html#masking-regions"><i class="fa fa-check"></i><b>6.5</b> Masking regions</a></li>
<li class="chapter" data-level="6.6" data-path="alignments.html"><a href="alignments.html#coregene"><i class="fa fa-check"></i><b>6.6</b> Core gene alignments</a></li>
<li class="chapter" data-level="6.7" data-path="alignments.html"><a href="alignments.html#snpalignment"><i class="fa fa-check"></i><b>6.7</b> SNP alignment</a></li>
<li class="chapter" data-level="6.8" data-path="alignments.html"><a href="alignments.html#vcf-conversion"><i class="fa fa-check"></i><b>6.8</b> VCF conversion</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="building-trees.html"><a href="building-trees.html"><i class="fa fa-check"></i><b>7</b> Building trees</a>
<ul>
<li class="chapter" data-level="7.1" data-path="building-trees.html"><a href="building-trees.html#fasttree"><i class="fa fa-check"></i><b>7.1</b> FastTree</a></li>
<li class="chapter" data-level="7.2" data-path="building-trees.html"><a href="building-trees.html#iq-tree"><i class="fa fa-check"></i><b>7.2</b> IQ-TREE</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="rooting-trees.html"><a href="rooting-trees.html"><i class="fa fa-check"></i><b>8</b> Rooting trees</a>
<ul>
<li class="chapter" data-level="8.1" data-path="rooting-trees.html"><a href="rooting-trees.html#outgroup"><i class="fa fa-check"></i><b>8.1</b> Outgroup</a></li>
<li class="chapter" data-level="8.2" data-path="rooting-trees.html"><a href="rooting-trees.html#midpoint"><i class="fa fa-check"></i><b>8.2</b> Midpoint</a></li>
<li class="chapter" data-level="8.3" data-path="rooting-trees.html"><a href="rooting-trees.html#molecular-clock"><i class="fa fa-check"></i><b>8.3</b> Molecular clock</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="rooting-trees.html"><a href="rooting-trees.html#treetime"><i class="fa fa-check"></i><b>8.3.1</b> TreeTime</a></li>
<li class="chapter" data-level="8.3.2" data-path="rooting-trees.html"><a href="rooting-trees.html#bactdating"><i class="fa fa-check"></i><b>8.3.2</b> BactDating</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bacterial Genomics Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genome-assembly" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Genome assembly<a href="genome-assembly.html#genome-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="spades" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Short-read<a href="genome-assembly.html#spades" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Illumina sequencing data can be assembled using <a href="https://github.com/ablab/spades">SPAdes</a>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="genome-assembly.html#cb14-1" tabindex="-1"></a><span class="co">#Paired end reads</span></span>
<span id="cb14-2"><a href="genome-assembly.html#cb14-2" tabindex="-1"></a><span class="ex">spades.py</span> <span class="at">-1</span> ERR11850183_1.fastq.gz <span class="at">-2</span> ERR11850183_2.fastq.gz <span class="at">-o</span> ERR11850183_assembly</span>
<span id="cb14-3"><a href="genome-assembly.html#cb14-3" tabindex="-1"></a><span class="co">#Single end reads</span></span>
<span id="cb14-4"><a href="genome-assembly.html#cb14-4" tabindex="-1"></a><span class="ex">spades.py</span> <span class="at">-s</span> ERR11850183.fastq.gz <span class="at">-o</span> ERR11850183_assembly</span>
<span id="cb14-5"><a href="genome-assembly.html#cb14-5" tabindex="-1"></a><span class="co">#Metagenomic assembly (paired reads only)</span></span>
<span id="cb14-6"><a href="genome-assembly.html#cb14-6" tabindex="-1"></a><span class="ex">metaspades.py</span> <span class="at">-1</span> ERR11850183_1.fastq.gz <span class="at">-2</span> ERR11850183_2.fastq.gz <span class="at">-o</span> ERR11850183_assembly</span></code></pre></div>
<p>The output <em>scaffolds.fasta</em> can then be used for further analyses. I would recommend moving and renaming this file.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="genome-assembly.html#cb15-1" tabindex="-1"></a><span class="fu">mv</span> ERR11850183_assembly/scaffolds.fasta ERR11850183.fasta</span></code></pre></div>
</div>
<div id="hybrid-and-long-read" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Hybrid and long-read<a href="genome-assembly.html#hybrid-and-long-read" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://github.com/rrwick/Unicycler?tab=readme-ov-file">Unicycler</a> can be used for long-read and hybrid assemblies.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="genome-assembly.html#cb16-1" tabindex="-1"></a><span class="co">#Hybrid assembly</span></span>
<span id="cb16-2"><a href="genome-assembly.html#cb16-2" tabindex="-1"></a><span class="ex">unicycler</span> <span class="at">-1</span> short_reads_1.fastq.gz <span class="at">-2</span> short_reads_2.fastq.gz <span class="at">-l</span> long_reads.fastq.gz <span class="at">-o</span> output_dir</span></code></pre></div>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Quality control<a href="genome-assembly.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://github.com/chklovski/CheckM2">CheckM2</a> can be used to estimate the completeness and contamination of an assembly. By default, CheckM2 assumes that assemblies end with the extension “fasta”.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="genome-assembly.html#cb17-1" tabindex="-1"></a><span class="ex">checkm2</span> predict <span class="at">--input</span> folder_containing_assemblies <span class="at">--output-directory</span> output_folder</span></code></pre></div>
<p><a href="https://github.com/ablab/quast">QUAST</a>, an alternative tool, can also be used for assembly quality control using a reference genome for comparison.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="genome-assembly.html#cb18-1" tabindex="-1"></a><span class="va">reference</span><span class="op">=</span>/path/to/reference/genome</span>
<span id="cb18-2"><a href="genome-assembly.html#cb18-2" tabindex="-1"></a><span class="co">#Assuming we are in a working directory containing assemblies with the file extension &quot;fasta&quot;</span></span>
<span id="cb18-3"><a href="genome-assembly.html#cb18-3" tabindex="-1"></a><span class="ex">quast.py</span> <span class="pp">*</span>fasta <span class="at">-R</span> <span class="va">$reference</span></span></code></pre></div>
</div>
<div id="prokka" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Annotation<a href="genome-assembly.html#prokka" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://github.com/tseemann/prokka">Prokka</a> is a fast method of genome annotation that is compatible with downstream pangenome analyses using panaroo.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="genome-assembly.html#cb19-1" tabindex="-1"></a><span class="ex">prokka</span> genome.fasta <span class="at">-o</span> output_folder_name</span></code></pre></div>
<p>You will then get a folder <em>output_folder_name</em> containing a number of files, including a gff file. I would recommend moving and renaming this file.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="genome-assembly.html#cb20-1" tabindex="-1"></a><span class="ex">prokka</span> ERR11850183.fasta <span class="at">-o</span> ERR11850183_annotation</span>
<span id="cb20-2"><a href="genome-assembly.html#cb20-2" tabindex="-1"></a><span class="fu">mv</span> ERR11850183_annotation/<span class="pp">*</span>gff ERR11850183.gff</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="fastq-quality-control.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="alignments.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/04-assembly.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
